packages/runtime/src/controls/controls/image.ts

```diff
@@ -0,0 +1,65 @@
+import { CopyContext, ReplacementContext } from '../../state/react-page'
+
+export type ImageControlData = string
+
+export const ImageControlType = 'makeswift::controls::image'
+
+export const ImageControlValueFormat = {
+  URL: 'makeswift::controls::image::format::url',
+  WithDimensions: 'makeswift::controls::image::format::with-dimensions',
+} as const
+
+type ImageControlValueFormat = typeof ImageControlValueFormat[keyof typeof ImageControlValueFormat]
+
+type ImageControlConfig = { label?: string; format?: ImageControlValueFormat }
+
+export type ImageControlDefinition<T extends ImageControlConfig = ImageControlConfig> = {
+  type: typeof ImageControlType
+  config: T
+}
+
+export function Image<T extends ImageControlConfig>(
+  config: T = {} as T,
+): ImageControlDefinition<T> {
+  return { type: ImageControlType, config }
+}
+
+Image.Format = ImageControlValueFormat
+
+export function imageCopy(value: ImageControlData, context: CopyContext): ImageControlData {
+  return context.replacementContext.fileIds.get(value) ?? value
+}
+
+if (import.meta.vitest) {
+  const { describe, test, expect } = import.meta.vitest
+
+  describe.concurrent('image copy', () => {
+    test('image is replaced by a one in replacement context', () => {
+      // Arrange
+      const data: ImageControlData = 'file-id'
+      const expected = JSON.parse(JSON.stringify(data).replace('file-id', 'testing'))
```

Comment:
Yeah, it is. I can change this. Was just doing this for big objects

Reply:
Got it. Even for that, the JSON stringify and back is a bit confusing, you can do object spread and stuff since that's a bit more explicit and clearer. The generic string `replace` seems a bit flakey